tests/pdb_test.go

```diff
@@ -0,0 +1,134 @@
+package tests_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	expect "github.com/google/goexpect"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	kvpointer "kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[sig-compute]Pod Disruption Budget (PDB)", decorators.SigCompute, func() {
+
+	It("should ensure the Pod Disruption Budget (PDB) is deleted when the VM is stopped via guest OS", func() {
+		var err error
+
+		By("Creating test VM")
+		vm := &v1.VirtualMachine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "testvm",
+				Namespace: testsuite.NamespaceTestDefault,
+			},
+			Spec: v1.VirtualMachineSpec{
+				RunStrategy: kvpointer.P(v1.RunStrategyManual),
+				Template: &v1.VirtualMachineInstanceTemplateSpec{
+					Spec: v1.VirtualMachineInstanceSpec{
+						EvictionStrategy: kvpointer.P(v1.EvictionStrategyLiveMigrate),
+						Domain: v1.DomainSpec{
+							Devices: v1.Devices{
+								Disks: []v1.Disk{
+									{
+										Name: "disk0",
+										DiskDevice: v1.DiskDevice{
+											Disk: &v1.DiskTarget{
+												Bus: "virtio",
+											},
+										},
+									},
+									{
+										Name: "cloudinitdisk",
+										DiskDevice: v1.DiskDevice{
+											Disk: &v1.DiskTarget{
+												Bus: "virtio",
+											},
+										},
+									},
+								},
+							},
+							Resources: v1.ResourceRequirements{
+								Requests: k8sv1.ResourceList{
+									k8sv1.ResourceMemory: resource.MustParse("128Mi"),
+								},
+							},
+						},
+						Volumes: []v1.Volume{
+							{
+								Name: "disk0",
+								VolumeSource: v1.VolumeSource{
+									ContainerDisk: &v1.ContainerDiskSource{
+										Image:           "registry:5000/kubevirt/cirros-container-disk-demo:devel",
+										ImagePullPolicy: k8sv1.PullIfNotPresent,
+									},
+								},
+							},
+							{
+								Name: "cloudinitdisk",
+								VolumeSource: v1.VolumeSource{
+									CloudInitNoCloud: &v1.CloudInitNoCloudSource{
+										UserDataBase64: "1234",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		By("Creating the VM")
+		_, err = kubevirt.Client().VirtualMachine(testsuite.NamespaceTestDefault).Create(context.Background(), vm, metav1.CreateOptions{})
```

Comment:
```suggestion
		vm, err = kubevirt.Client().VirtualMachine(testsuite.NamespaceTestDefault).Create(context.Background(), vm, metav1.CreateOptions{})
```
Better to update the `vm` with the created one.

Reply:
agreed